ccm_web/server/src/api/api.interfaces.ts

```diff
@@ -28,3 +33,16 @@ export interface APIErrorData {
 export function isAPIErrorData (value: unknown): value is APIErrorData {
   return hasKeys(value, ['statusCode', 'errors'])
 }
+
+export interface ExternalEnrollmentUserData {
+  [email: string]: {
+    userCreated?: CanvasUserLoginEmail | APIErrorData | false
+    inviteResult?: CirrusInvitationResponse | APIErrorData
+    enrollment?: CanvasEnrollment | APIErrorData
```

Comment:
Yeah, this is something I want to look at, I just marked them all optional to get it working. We can definitely make `userCreated` required. Basically because we're accumulating, we either have to leave some things optional as we build it, or we do some other object construction where we hold parts until everything is ready. With TypeScript, it's tricky to change objects too much unless you type it at the beginning. I have to play around with it more.

Reply:
`userCreated` is now required.